assets/prometheus-k8s/rules.yaml

```diff
@@ -1873,7 +1889,7 @@ spec:
           sum by (job) (up{job=~".*etcd.*"} == bool 0)
         or
           count by (job,endpoint) (
-            sum by (job,endpoint,To) (rate(etcd_network_peer_sent_failures_total{job=~".*etcd.*"}[3m])) > 0.01
+            sum by (job,endpoint,To) (rate(etcd_network_peer_sent_failures_total{job=~".*etcd.*"}[1m])) > 0.01
```

Comment:
This is doing rate over 1m, but etcd scrape interval is set to 30s. Such setup may result in incorrect data as you need at least 3 data points to do `rate()`. Either etcd scrape interval needs to be fixed to 15s or rate needs to be done over at least 2m.

Reply:
Good catch, thanks. My gut reaction is to shorten the scrape interval in this case, but I'd like some feedback from others. cc @hexfusion @wking @smarterclayton 

Makes me wonder if my testing was somehow flawed...

Reply:
Are there any benefits to a more granular scrape interval for etcd specifically? Having a higher resolution seems to be useful in this specific case (and I'm going to do more testing to see if the same results can be achieved with 30s+2m window), but I'm curious what other reasons there might be to change the interval.

Reply:
Here's what a reboot looks like with 30s and 15s etcd scrape intervals

<img width="1664" alt="Screen Shot 2020-07-16 at 10 32 55 AM" src="https://user-images.githubusercontent.com/298299/87684177-eebe1580-c74f-11ea-81f5-8a011be8f62c.png">

Reply:
I don't see any immediate benefit of having higher resolution metrics for etcd as all alerts (apart from this one) use longer range selector and additionally have `for` statements.

As for this alert, what about shortening `for` statement and increasing range selector to `2m`?

Reply:
Revisiting this, @wking made the point that when this alert accurately fires and someone gets paged, an extra 5 or 10 minutes before being notified is unlikely to be a significant factor. And the alert today isn't actually considered actionable. Further, I'm also observing a ton of variance during reboots with regards to `up` and `etcd_network_peer_sent_failures_total` leveling off — anywhere from 1 to 6 minutes.

With those considerations, what if we used a 2m window and increased the `for` clause to something far more conservative like 10m? It seems like that could yield the accuracy improvements at the expense of a (hopefully insignificant) delay in alerting.

Reply:
https://github.com/etcd-io/etcd/pull/12177